subsys/dfu/boot/mcuboot.c

```diff
@@ -96,15 +107,59 @@ uint8_t boot_fetch_active_slot(void)
 	}
 
 	LOG_DBG("Active slot: %d", slot);
+	/* Map slot number back to flash area ID */
+	switch (slot) {
+	case 0:
+		fa_id = FIXED_PARTITION_ID(SLOT0_PARTITION);
+		break;
+
+#if FIXED_PARTITION_EXISTS(SLOT1_PARTITION)
+	case 1:
+		fa_id = FIXED_PARTITION_ID(SLOT1_PARTITION);
```

Comment:
this change is a breaking change, if people implemented the former behavior in their own compatible with mcuboot bootloader, should have gotten a option to preserve previous behavior, as changing this in their bootloader and getting that fixed bootloader on already shipped products, is often not possible, as bootloaders are not meant to be upgraded

Reply:
I guess you had some logic to map the flash area ID in your application? Could you use `BLINFO_BOOTLOADER_VERSION` to determine which version of MCUBoot is executing on your platform? Failing that, I have no issue with a Kconfig to use the previous functionality here